File: Stacks&Queues/Queue.js
class Queue {
    constructor() {
        this.arr = [];
        this.itemcount = 0;
    }
    
    enqueue(item) {
        this.arr.push(item);
        this.itemcount++;
    }

    dequeue() {
        this.arr.shift()
    }

    empty() {
        return this.arr.length === 0;
    }

    top() {
        return this.arr[this.arr.length - 1];
    }

    display() {
        var str = ""
        for(var i = 0; i < this.arr.length; i++) {
            str += this.arr[i] + " ";
        }
        console.log(str)
    }
}

let q = new Queue();
q.enqueue(1)
q.enqueue(2)
q.enqueue(3)
q.display()
q.dequeue()
q.display()

File: Stacks&Queues/Queue.py
class Queue:
    def __init__(self):
        self.arr = []
        self.extra = []
        
    def enqueue(self,item):
        self.arr.append(item)
        
    def dequeue(self):
        if(not self.empty()):
            while not self.empty():
                temp = self.arr.pop()
                self.extra.append(temp)
            self.extra.pop()
            self.arr = self.extra[::-1]
                
    def empty(self):
        return len(self.arr) == 0
    
    def top(self):
        return self.arr[len(self.arr) - 1]
    
    def display(self):
        print(self.arr)
        
        
if __name__ == "__main__":
    q = Queue()
    q.enqueue(1)
    q.enqueue(2)
    q.enqueue(3)
    q.display()
    q.dequeue()
    q.display()
    